compiler/rustc_infer/src/lib.rs

```diff
@@ -23,6 +23,8 @@
 #![feature(never_type)]
 #![feature(try_blocks)]
 #![recursion_limit = "512"] // For rustdoc
+// #![deny(rustc::untranslatable_diagnostic)]
+// #![deny(rustc::diagnostic_outside_of_impl)]
```

Comment:
What is still remaining that prevents these from being removed?

Reply:
There is still quite a bit to do - about 60 lints to resolve(54 of which are in infer/error_reporting/mods.rs). Also sometimes this api is abstracted in a function, like `note_and_explain_region`, so there is even more than that.

Reply:
Do you want to land some of this work before the crate is complete or just keep going until it's all done?

Reply:
Should be fine to merge.

Reply:
Could we remove these commented lines before doing so?